cdap-ui/app/cdap/components/SecureKeys/SecureKeyEdit/index.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import withStyles, { StyleRules, WithStyles } from '@material-ui/core/styles/withStyles';
+import TextField from '@material-ui/core/TextField';
+import { MySecureKeyApi } from 'api/securekey';
+import classnames from 'classnames';
+import WidgetWrapper from 'components/ConfigurationGroup/WidgetWrapper';
+import { SecureKeyStatus } from 'components/SecureKeys';
+import { COMMON_DELIMITER, COMMON_KV_DELIMITER } from 'components/SecureKeys/constants';
+import { Map } from 'immutable';
+import isNil from 'lodash/isNil';
+import React from 'react';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+
+const styles = (theme): StyleRules => {
+  return {
+    margin: {
+      margin: `${theme.Spacing(3)}px ${theme.spacing(1)}px`,
+    },
+    textField: {
+      width: '45ch',
+    },
+    keyvalueField: {
+      width: '60ch',
+    },
+  };
+};
+
+interface ISecureKeyEditProps extends WithStyles<typeof styles> {
+  state: any;
+  dispatch: React.Dispatch<any>;
+  open: boolean;
+  handleClose: () => void;
+}
+
+const SecureKeyEditView: React.FC<ISecureKeyEditProps> = ({
+  classes,
+  state,
+  dispatch,
+  open,
+  handleClose,
+}) => {
+  const { secureKeys, activeKeyIndex } = state;
+
+  const keyMetadata = secureKeys.get(activeKeyIndex);
+  const keyID = keyMetadata ? keyMetadata.get('name') : '';
+  const [localDescription, setLocalDescription] = React.useState(
+    keyMetadata ? keyMetadata.get('description') : ''
+  );
+  const [localData, setLocalData] = React.useState(keyMetadata ? keyMetadata.get('data') : '');
```

Comment:
is this the actual data of the secure key? we should not be displaying the existing key. The edit modal should only allow the user to change the value, not to retrieve it

Reply:
Resolved